ui/src/criteria/attribute.tsx

```diff
@@ -81,73 +136,216 @@ function AttributeEdit(props: AttributeEditProps) {
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMin = event.target.value === "" ? "" : Number(event.target.value);
+      setMinInputValue(newMin); // Make sure empty input won't get changed.
+      newMin = typeof newMin === "number" ? newMin : 0;
+      const maxValue = props.data?.max || maxBound;
 
-  const selectionIndex = (hint: tanagra.EnumHintValue) =>
-    props.data.selected.findIndex(
-      (row) => row.id === hint.attributeValue?.int64Val
+      newMin = Math.max(minBound, newMin);
+      newMin = Math.min(maxValue, newMin);
+      updateValue(newMin, maxValue);
+    };
+    const handleMinInputBlur = () => {
+      let newMin = typeof minInputValue === "number" ? minInputValue : 0;
+      const maxValue = props.data?.max || maxBound;
+      newMin = Math.max(newMin, minBound);
+      newMin = Math.min(newMin, maxValue);
+      setMinInputValue(newMin);
+    };
+
+    const handleMaxInputChange = (
+      event: React.ChangeEvent<HTMLInputElement>
+    ) => {
+      let newMax = event.target.value === "" ? "" : Number(event.target.value);
+      setMaxInputValue(newMax); // Make sure empty input won't get changed.
+      newMax = typeof newMax === "number" ? newMax : 0;
+      const minValue = props.data?.min || minBound;
+
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      updateValue(minValue, newMax);
+    };
+    const handleMaxInputBlur = () => {
+      let newMax = typeof maxInputValue === "number" ? maxInputValue : 0;
+      const minValue = props.data?.min || minBound;
+      newMax = Math.max(minValue, newMax);
+      newMax = Math.min(newMax, maxBound);
+      setMaxInputValue(newMax);
+    };
+
+    return (
+      <Box sx={{ width: "30%", minWidth: 300, margin: 2 }}>
+        <Grid container spacing={2}>
+          <Grid item>
+            <Input
+              value={minInputValue}
+              size="medium"
+              onChange={handleMinInputChange}
+              onBlur={handleMinInputBlur}
+              inputProps={{
+                step: ((maxBound - minBound) / 20) | 0,
+                min: minBound,
+                max: maxBound,
+                type: "number",
+                "aria-labelledby": "input-slider",
+              }}
+            />
+          </Grid>
+          <Grid item xs>
+            <Slider
+              value={[props.data?.min || minBound, props.data?.max || maxBound]}
+              onChange={handleChange}
+              valueLabelDisplay="auto"
+              getAriaValueText={(value) => value.toString()}
+              min={minBound}
+              max={maxBound}
+              disableSwap
+            />
+          </Grid>
+          <Grid item>
+            <Input
+              value={maxInputValue}
+              size="small"
+              onChange={handleMaxInputChange}
+              onBlur={handleMaxInputBlur}
+              inputProps={{
+                step: Math.floor((maxBound - minBound) / 20),
```

Comment:
I think this won't do anything once the input type isn't "number" anymore but you should check.

Reply:
I think it works. there is no decimal.